tap_freshdesk/client.py

```diff
@@ -0,0 +1,83 @@
+import time
+import backoff
+import requests
+import singer
+from singer import utils
+
+
+LOGGER = singer.get_logger()
+BASE_URL = "https://{}.freshdesk.com"
+DEFAULT_PAGE_SIZE = 100
+
+
+class FreshdeskClient:
+    """
+    The client class is used for making REST calls to the Freshdesk API.
+    """
+
+    def __init__(self, config):
+        self.config = config
+        self.session = requests.Session()
+        self.base_url = BASE_URL.format(config.get("domain"))
+        self.page_size = self.get_page_size()
+
+    def __enter__(self):
+        self.check_access_token()
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        # Kill the session instance.
+        self.session.close()
+
+    def get_page_size(self):
+        """
+        This function will get page size from config,
+        and will return the default value if an invalid page size is given.
+        """
+        page_size = self.config.get('page_size')
```

Comment:
```suggestion
        page_size = self.config.get('page_size', DEFAULT_PAGE_SIZE)
```
We can remove the below condition - 
`if page_size is None:
            return DEFAULT_PAGE_SIZE`